api/v1beta1/azureclusteridentity_types.go

```diff
@@ -59,6 +59,9 @@ type AzureClusterIdentitySpec struct {
 	// ClientSecret is a secret reference which should contain either a Service Principal password or certificate secret.
 	// +optional
 	ClientSecret corev1.SecretReference `json:"clientSecret,omitempty"`
+	// certPath is the path where certificates exist. When set, it takes precedence over ClientSecret for types that uses certs like ServicePrincipalCertificate.
+	// +optional
+	CertPath string `json:"certPath,omitempty"`
```

Comment:
For each AzureClusterIdentity, CAPZ also creates a Secret for ASO to use to authenticate, so that flow should also be updated here: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/976f89385bb1efb7f80ccaeda03142916bc82f09/controllers/asosecret_controller.go#L249

I don't think ASO has a similar "read cert from path" mechanism, so this would have to still dump the certificate data into a Secret because that's all ASO knows how to read: https://azure.github.io/azure-service-operator/guide/authentication/credential-format/#service-principal-using-a-client-certificate. Adding this functionality there first might be a better approach if that's mainly what you're trying to avoid.

If you're not seeing auth errors on ASO resources when you test this though that would be interesting.

Reply:
Thanks Jon, I'll take a look at this.

FWIW - we are wanting this from a self-managed CAPZ perspective.